src/psyclone/domain/lfric/algorithm/alg_gen.py

```diff
@@ -0,0 +1,477 @@
+    # respective function spaces extracted from the kernel metadata.
+    for sym, space in kern_args.fields:
+        initialise_field(prog, sym, space)
+
+    for qr_sym, shape in kern_args.quadrature_objects:
+        initialise_quadrature(prog, qr_sym, shape)
+
+    return kern_args
+
+
+def generate(kernel_path):
+    '''
+    Generates LFRic algorithm code that calls the supplied kernel through
+    an 'invoke'. All of the arguments required by the kernel are constructed
+    and intialised appropriately. Fields and scalars are all set to unity.
+
+    :param str kernel_path: location of Kernel source code.
+
+    :returns: Fortran algorithm code.
+    :rtype: str
+
+    :raises NotImplementedError: if the specified kernel file does not \
+        follow the LFRic naming convention by having a module with a name \
+        ending in '_mod'.
+
+    '''
+    # Create PSyIR for a skeleton driver routine.
+    prog = _create_alg_driver("lfric_alg", 20)
+    table = prog.symbol_table
+
+    # Parse the kernel metadata (this still uses fparser1 as that's what
+    # the meta-data handling is currently based upon).
+    parse_tree = get_kernel_parse_tree(kernel_path)
+
+    # Get the name of the module that contains the kernel and create a
+    # ContainerSymbol for it.
+    kernel_mod_name = parse_tree.content[0].name
+    # TODO #1453. The current meta-data parsing requires that we specify
+    # the name of the kernel. It would be much better if we could query the
+    # meta-data for the name of the kernel. For now we require that the LFRic
+    # naming scheme is strictly adhered to (since this is simpler than trying
+    # to walk through the deprecated fparser1 parse tree).
+    if not kernel_mod_name.endswith("_mod"):
+        raise NotImplementedError(
+            f"The supplied kernel ({kernel_path}) contains a module named "
+            f"'{kernel_mod_name}' which does not follow the LFRic naming "
+            f"convention of ending in '_mod'.")
+    kernel_name = kernel_mod_name[:-4] + "_type"
+
+    kernel_mod = table.new_symbol(kernel_mod_name, symbol_type=ContainerSymbol)
+    kernel_routine = table.new_symbol(kernel_name,
+                                      symbol_type=DataTypeSymbol,
+                                      datatype=DeferredType(),
+                                      interface=ImportInterface(kernel_mod))
+
+    ktype = KernelTypeFactory(api="dynamo0.3").create(parse_tree,
+                                                      name=kernel_name)
+    # Construct a DynKern using the metadata. This is used when constructing
+    # the kernel argument list.
+    kern = DynKern()
+    kern.load_meta(ktype)
+
+    # Declare and initialise the data structures required by the kernel
+    # arguments. Appropriate symbols are added to the symbol table associated
+    # with the routine we are constructing.
+    kern_args = construct_kernel_args(prog, kern)
+
+    # Initialise argument values to unity.
```

Comment:
As we do elsewhere (albeit I can't remember precisely where) I'm being lazy and just relying on the compiler/run-time to do the right type coercion. Since this is just setting everything to unity this is fine. Obviously, if we were doing something more sophisticated then it wouldn't be. I'll add a comment about this.

Reply:
Agreed.